plugins/rbac/src/components/CreateRole/RoleForm.tsx

```diff
@@ -72,12 +72,13 @@ export const RoleForm = ({
   const navigate = useNavigate();
   const rbacApi = useApi(rbacApiRef);
 
-  const navigateTo = (action?: string) => {
-    const stateProp = action
-      ? { state: { toastMessage: `Role ${action} successfully` } }
-      : { state: { toastMessage: '' } };
-    if (step && roleName) {
-      const { kind, namespace, name } = getKindNamespaceName(roleName);
+  const navigateTo = (rName?: string, action?: string) => {
+    const stateProp =
+      rName && action
+        ? { state: { toastMessage: `Role ${rName} ${action} successfully` } }
+        : { state: { toastMessage: '' } };
```

Comment:
Couldn't the stateProp be undefined when there is no action instead of an empty (string) toastMessage?